### ISSUE TITLE
Callback in jsdoc

### ISSUE THREAD

**Issue by clitetailor** [Microsoft/vscode#15576]
- VSCode Version: 1.7.1
- OS Version: Windows 10

Sorry, my English is quite poor. I tried to add callback to jsdoc without success.

```
@param {{ log: (route: number[]) => void }} res
```

So i have to do this
```
@param {{log : function(number[]) => void }} res
```

It's nearly the same but not exact what i want since `log: (route:number[]) => void` become `log: (p0: number[]) => void`. Can you add this feature to jsdoc. Thanks!

**Comment by ramya-rao-a** [comment]
@clitetailor I don't understand what the issue is here.

So what happens when you add `@param {{ log: (route: number[]) => void }} res` ?

**Comment by clitetailor** [comment]
the abbreviation does not show exact what i want :(

![image](https://cloud.githubusercontent.com/assets/16368559/20397948/11507566-ad1e-11e6-916d-8bb39e13517e.png)

**Comment by mjbvz** [comment]
This has been fixed on the TS side in ts 2.2 dev builds at least:

![nov-22-2016 11-46-30](https://cloud.githubusercontent.com/assets/12821956/20539241/3380b842-b0a9-11e6-99a9-4c3ce124bc9e.gif)

We have not picked up this in VSCode, even in insiders builds yet, so I'll keep this issue open for now. Until then, you can try installing one of the nightly typescript builds and pointing VSCode to use it using the`typescript.tsdk` setting.

**Comment by mjbvz** [comment]
This appears to be fixed in typescript `typescript@2.1.4-insiders.20161201` which we just adopted. It should be available in the next insiders build and in VSCode 1.8
